analyzers/src/SonarAnalyzer.Common/Common/DisjointSets.cs

```diff
@@ -0,0 +1,47 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Common;
+
+/// <summary>
+/// Data structure for working with disjoint sets of strings, to perform union-find operations with equality semantics:
+/// i.e. reflexivity, symmetry and transitivity. See https://en.wikipedia.org/wiki/Disjoint-set_data_structure.
+///
+/// An example of use is to build undirected connected components of dependencies, where each node is the identifier.
+///
+/// Uses a dictionary of strings as a backing store. The dictionary represents a forest of trees, where each node is
+/// a string and each tree is a set of nodes.
+/// </summary>
+public class DisjointSets
```

Comment:
I would add a link to the Disjoint set documentation.
For example I found this super helpful: https://www.geeksforgeeks.org/introduction-to-disjoint-set-data-structure-or-union-find-algorithm/

Once I read this, I understood why you set these names, why these methods, etc. (this is an example link, please link the one you think it's best).

Reply:
The link to Wikipedia is specified just two lines above.
I have added the link to geekforgeeks.org as well.